### PR TITLE
Update build system and docs to state cuda 11 required

### PR DESCRIPTION
**Pull request by robertmaynard**
closes https://github.com/rapidsai/rmm/issues/736

rmm-0.20 increase the minimum requirements in the following way:

- GCC version 9.0+ is required
- CUDA and C++ code now is compiled with `-std=c++17`
- We require CUDA Toolkit version 11.0 or greater

This updates the build-system and the README with these new requirements

### CODE REVIEW

**Comment by robertmaynard** [comment]
rerun tests

**Comment by robertmaynard** [comment]
It is expected that all the builds won't pass since we still have CUDA 10.1 and 10.2 testers

**Comment by harrism** [comment]
We still haven't heard from 2 of our stakeholders on #736...

File: README.md [hunk 1]
```diff
@@ -54,14 +54,14 @@ See the [Get RAPIDS version picker](https://rapids.ai/start.html) for more OS an
 
 Compiler requirements:
 
-* `gcc`     version 7.0 or higher required
-* `nvcc`    version 9.0 or higher recommended
+* `gcc`     version 9.0 or higher required
```
[the remark this answers is not in the document]
Comment (harrism) [reply README.md]: BTW, we *do* use nvcc to compile some tests/benchmarks.

[the remark this answers is not in the document]
Comment (robertmaynard) [reply README.md]: The inline comment is hiding the ```+ * `nvcc`    version 11.0+ ``` line

**Comment by robertmaynard** [comment]
rerun tests

**Comment by harrism** [comment]
> We still haven't heard from 2 of our stakeholders on #736...

We have now. Good to go.

**Review by harrism — APPROVED** [review]

**Review by kkraus14 — APPROVED** [review]

**Comment by kkraus14** [comment]
rerun tests

**Comment by kkraus14** [comment]
@gpucibot merge

**Pull request merged**